### PR TITLE
match patterns at tile boundaries

### PR DESCRIPTION
**Pull request by kelvinabrokwa**
From:

actual

To: 

expected

- [x] add tests

### CODE REVIEW

**Comment by kelvinabrokwa** [comment]
@ansis @lucaswoj @jfirebaugh :eyes:

File: js/render/draw_fill.js [hunk 1]
```diff
@@ -135,18 +135,38 @@ function drawFill(painter, source, layer, coord) {
...
+            1 / (imageSizeScaledB[0] * factor),
+            1 / (imageSizeScaledB[1] * factor)
         ]);
 
+        // shift images to match at tile boundaries
+        var offsetAx = ((tile.tileSize % imageSizeScaledA[0]) * tile.coord.x) / imageSizeScaledA[0];
+        var offsetAy = ((tile.tileSize % imageSizeScaledA[1]) * tile.coord.y) / imageSizeScaledA[1];
+
+        var offsetBx = ((tile.tileSize % imageSizeScaledB[0]) * tile.coord.x) / imageSizeScaledB[0];
+        var offsetBy = ((tile.tileSize % imageSizeScaledB[1]) * tile.coord.y) / imageSizeScaledB[1];
```
Comment (lucaswoj) [inline js/render/draw_fill.js]: You can get seamless tiling over the dateline by including the 'coord.w' property in these calculations

**Comment by jfirebaugh** [comment]
:+1:

File: js/render/draw_fill.js (hunk 1 above)
Comment (ansis) [inline js/render/draw_fill.js]: good idea. I has the downside that the pattern will jump once when you pan over the dateline when the coordinates are converted to the -180 to 180 range. The jump may be less visible than the seam though

Comment (lucaswoj) [inline js/render/draw_fill.js]: I'm sure we can math our way around that jump (worst case, there's a different branch for `coord.w < 0`) :heavy_division_sign:

Comment (ansis) [inline js/render/draw_fill.js]: I thought we were normalizing lat/lng to -180..180 but it turns out we aren't. We should definitely do seamless tiling over the dateline then

Comment (kelvinabrokwa) [inline js/render/draw_fill.js]: yup! Change the above to 

```js
var offsetBx = ((tile.tileSize % imageSizeScaledB[0]) * (tile.coord.x + coord.w * Math.pow(2, tile.coord.z))) / imageSizeScaledB[0];
```
and we're in beezwax!

Thanks @lucaswoj!

**Comment by ansis** [comment]
great! -native?

**Comment by kelvinabrokwa** [comment]
@ansis yup, hitting that now -> https://github.com/mapbox/mapbox-gl-native/issues/3476

**Comment by skorasaurus** [comment]
\o/  can't wait to use this!

**Pull request merged**